source/routing/isochron.cpp

```diff
@@ -116,32 +117,60 @@ type::Polygon circle(const type::GeographicalCoord& center,
     return points;
 }
 
+static type::MultiPolygon merge_poly(const type::MultiPolygon& multi_poly, type::Polygon poly) {
+    type::MultiPolygon multi_polys_merged;
+    for(const type::Polygon& p: multi_poly) {
+        if (boost::geometry::intersects(p, poly)) {
+           type::MultiPolygon poly_union;
+           boost::geometry::union_(poly, p, poly_union);
+           poly = poly_union[0];
+        } else {
+            multi_polys_merged.push_back(p);
+        }
+    }
+    multi_polys_merged.push_back(poly);
+    return multi_polys_merged;
+}
+
+template<typename T>
+static bool in_bound(const T & begin, const T & end, bool clockwise) {
+    return (clockwise && begin < end) ||
+            (!clockwise && begin > end);
+}
+
 type::MultiPolygon build_isochron(RAPTOR& raptor,
                                 const std::vector<type::StopPoint*>& stop_points,
                                 const bool clockwise,
-                                const DateTime& init_dt,
                                 const DateTime& bound,
-                                const map_stop_point_duration& origin) {
+                                const map_stop_point_duration& origin,
+                                const double& speed,
+                                const int& max_duration) {
     type::MultiPolygon circles;
-    double speed = 0.8; // About 3km/h TODO pass it in param
     const auto& data_departure = raptor.data.pt_data->stop_points;
     for (auto it = origin.begin(); it != origin.end(); ++it){
-        int duration_max = abs(int(bound) - (abs(int(init_dt) - int(it->second.total_seconds()))));
-        const auto& center = data_departure[it->first.val]->coord;
-        circles.push_back(circle(center , duration_max * speed));
+        if (it->second.total_seconds() < max_duration) {
+            int duration_max = max_duration - int(it->second.total_seconds());
+            //Delete the circles too small to avoid rounding error
+            if (duration_max * speed < 5) {continue;}
```

Comment:
I don't like this magic number, maybe you can define it globally with a nice name ?